### PR TITLE
Add ntddstor definitions

### PR DESCRIPTION
**Pull request by docbrown**
<del>Needs #372 merged first.</del>

### CODE REVIEW

**Comment by docbrown** [comment]
Well, I sure messed that up, didn't I?

File: src/shared/ntddstor.rs [hunk 1]
```diff
@@ -0,0 +1,1843 @@
...
+use shared::devpropdef::DEVPROPKEY;
+use shared::guiddef::GUID;
+use shared::minwindef::{UCHAR, ULONG, USHORT};
+use shared::ntdef::{CHAR, LARGE_INTEGER, LONGLONG, MAXULONG, SHORT, ULONGLONG, WCHAR};
+use um::winioctl::{
+    DEVICE_TYPE, FILE_ANY_ACCESS, FILE_DEVICE_MASS_STORAGE, FILE_READ_ACCESS,
+    FILE_WRITE_ACCESS, METHOD_BUFFERED,
+};
+use um::winnt::{ANYSIZE_ARRAY, BOOLEAN};
+DEFINE_GUID!(GUID_DEVINTERFACE_DISK,
```
Comment (retep998) [inline src/shared/ntddstor.rs:17]: These are item level macros and should use `{}` instead of `()`.

File: src/shared/ntddstor.rs [hunk 2]
```diff
@@ -0,0 +1,1843 @@
...
+    BusTypeNvme,
+    BusTypeSCM,
+    BusTypeUfs,
+    BusTypeMax,
+    BusTypeMaxReserved = 0x7F,
+}}
+pub type PSTORAGE_BUS_TYPE = *mut STORAGE_BUS_TYPE;
+#[inline]
+pub fn SupportsDeviceSharing(BusType: STORAGE_BUS_TYPE) -> bool {
+    (BusType == BusTypeScsi)  ||
```
Comment (retep998) [inline src/shared/ntddstor.rs:357]: Binary operators should be at the beginning of the line, not the end.

File: src/shared/ntddstor.rs [hunk 3]
```diff
@@ -0,0 +1,1843 @@
...
+    Command: [UCHAR; TELEMETRY_COMMAND_SIZE],
+    StartTime: ULONGLONG,
+    EndTime: ULONGLONG,
+    OperationStatus: ULONG,
+    OperationError: ULONG,
+    StackSpecific: DEVICEDUMP_STORAGESTACK_PUBLIC_STATE_RECORD_StackSpecific,
+}}
+pub type PDEVICEDUMP_STORAGESTACK_PUBLIC_STATE_RECORD =
+    *mut DEVICEDUMP_STORAGESTACK_PUBLIC_STATE_RECORD;
+impl DEVICEDUMP_STORAGESTACK_PUBLIC_STATE_RECORD {
```
Comment (retep998) [inline src/shared/ntddstor.rs:1562]: I'd rather not have these methods defined for now.

**Comment by nico-abram** [comment]
Hi @docbrown are you still interested in getting this merged?

The lack of activity and the still-open review comments make me believe this can be closed but I wanted to ask

**Pull request closed without merging**